mfr/providers/osf/provider.py

```diff
@@ -66,23 +66,24 @@ def __init__(self, request, url):
             metadata_url = download_url.replace('/file?', '/data?', 1)
             metadata_request = await self._make_request('GET', metadata_url)
             metadata = await metadata_request.json()
+            await metadata_request.release()
```

Comment:
It's been a while. I think I'm going to be saying that to you a lot now that you've taken on these ancient PRs. Anyway, I believe this is simply because it has become our standard to manually close all requests. I believe this is because we were getting a ton of "unclosed" type errors in sentry. @felliott correct me if I'm wrong here.

Reply:
Np, I understand. That's why I try to ask as many questions to help us recall the decision made long time ago, double check a change and document it somewhere. Hope it is constructive rather than causing overhead for you guys.

I agree that it is better we explicitly close/release the connection.